minio/minio
# Add support for bucket encryption feature

## Description
This PR adds support for default encryption for buckets (SSE-S3 only).
Fixes #8714
## Motivation and Context
To allow users take advantage of configuring default encryption for buckets on MinIO.
(ref: https://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html)

## How to test this PR?
1. Put bucket encryption configuration for bucket `foo`
```shell
AWS_ACCESS_KEY_ID=minio AWS_SECRET_ACCESS_KEY=minio123 aws --endpoint-url http://localhost:9001 s3api put-bucket-encryption --bucket foo --server-side-encryption-configuration '{"Rules": [{"ApplyServerSideEncryptionByDefault": {"SSEAlgorithm": "AES256"}}]}' --content-md5 5bZurju84n5lUGMQ6mrGXQ==
```
2. Get bucket encryption configuration for bucket `foo`
```shell
AWS_ACCESS_KEY_ID=minio AWS_SECRET_ACCESS_KEY=minio123 aws --endpoint-url http://localhost:9001 s3api get-bucket-encryption --bucket foo 
{
    "ServerSideEncryptionConfiguration": {
        "Rules": [
            {
                "ApplyServerSideEncryptionByDefault": {
                    "SSEAlgorithm": "AES256"
                }
            }
        ]
    }
}
```
3. Delete bucket encryption configuration for bucket `foo`
```shell
AWS_ACCESS_KEY_ID=minio AWS_SECRET_ACCESS_KEY=minio123 aws --endpoint-url http://localhost:9001 s3api delete-bucket-encryption --bucket foo
```

4. Perform a get operation to observe 'not found' error
```shell
AWS_ACCESS_KEY_ID=minio AWS_SECRET_ACCESS_KEY=minio123 aws --endpoint-url http://localhost:9001 s3api get-bucket-encryption --bucket foo

An error occurred (ServerSideEncryptionConfigurationNotFoundError) when calling the GetBucketEncryption operation: The bucket encryption configuration does not exist
```

## Types of changes

## Checklist:
